system_tests/nodeinterface_test.go

```diff
@@ -1,20 +1,143 @@
 // Copyright 2021-2022, Offchain Labs, Inc.
 // For license information, see https://github.com/nitro/blob/master/LICENSE
 
+// race detection makes things slow and miss timeouts
+//go:build !race
+// +build !race
+
 package arbtest
 
 import (
 	"context"
 	"math/big"
 	"testing"
 
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/offchainlabs/nitro/arbnode"
 	"github.com/offchainlabs/nitro/arbos/util"
 	"github.com/offchainlabs/nitro/solgen/go/node_interfacegen"
 )
 
+func callFindBatchContainig(t *testing.T, ctx context.Context, client *ethclient.Client, nodeAbi *abi.ABI, blockNum uint64) uint64 {
+	findBatch := nodeAbi.Methods["findBatchContainingBlock"]
+	callData := append([]byte{}, findBatch.ID...)
+	packed, err := findBatch.Inputs.Pack(blockNum)
+	Require(t, err)
+	callData = append(callData, packed...)
+	msg := ethereum.CallMsg{
+		To:   &types.NodeInterfaceAddress,
+		Data: callData,
+	}
+	returnData, err := client.CallContract(ctx, msg, nil)
```

Comment:
Are we not able to use abigen for this? Why can't we do NewNodeInterface and call the methods that way? Same goes for getL1Confirmations.